go/vt/vtgate/planbuilder/operators/horizon_planning.go

```diff
@@ -673,47 +441,182 @@ func setUpperLimit(in *Limit) (ops.Operator, *rewrite.ApplyResult, error) {
 	return in, rewrite.SameTree, nil
 }
 
-func tryPushingDownLimitInRoute(in *Limit, src *Route) (ops.Operator, *rewrite.ApplyResult, error) {
-	if src.IsSingleShard() {
-		return rewrite.Swap(in, src, "limit pushed into single sharded route")
+func tryPushingDownOrdering(ctx *plancontext.PlanningContext, in *Ordering) (ops.Operator, *rewrite.ApplyResult, error) {
+	switch src := in.Source.(type) {
+	case *Route:
+		return rewrite.Swap(in, src, "push ordering under route")
```

Comment:
hmm... shouldn't we check whether this is a single shard route before pushing down? This doesn't look correct